src/matter/interaction/InteractionServer.ts

```diff
@@ -168,6 +171,74 @@ export class InteractionServer implements ProtocolHandler<MatterDevice> {
         };
     }
 
+    handleWriteRequest(exchange: MessageExchange<MatterDevice>, {suppressResponse, writeRequests }: WriteRequest): WriteResponse | undefined {
+        logger.debug(`Received write request from ${exchange.channel.getName()} for ${writeRequests.length} attributes, suppressResponse=${suppressResponse}`);
+
+        // TODO consider TimedRequest constrains
+
+        const writeResponses = [];
+        for (const request of writeRequests) {
+            const attributes = this.getAttribute({ endpointId: request.path.endpointId, clusterId: request.path.clusterId, id: request.path.attributeId});
+            if (attributes.length === 1) {
+                // TODO add checks or dataVersion
+                try {
+                    const data = attributes[0].attribute.schema.decodeTlv(request.data);
+                    logger.debug(`Handle write request from ${exchange.channel.getName()} resolved to: ${this.resolveAttributeName(attributes[0].path)}=${Logger.toJSON(data)} (${request.dataVersion})`);
+                    attributes[0].attribute.set(data, exchange.session);
+                } catch (e: any) {
+                    logger.error(`Error while handling write request from ${exchange.channel.getName()}: ${e.message}`);
+                    writeResponses.push({
+                        status: {
+                            status: StatusCode.ConstraintError
+                        },
+                        path: attributes[0].path,
+                    });
+                }
+            } else if (attributes.length === 0) {
+                logger.error(`Attribute ${this.resolveAttributeName({ endpointId: request.path.endpointId, clusterId: request.path.clusterId, id: request.path.attributeId })} not found`);
+                writeResponses.push({
+                    status: {
+                        status: StatusCode.UnsupportedWrite
+                    }, // TODO: Find correct status code
+                    path: request.path,
+                });
+            } else {
+                attributes.forEach(({ path, attribute }) => {
+                    // Todo respect ACL and TimedWrite flags
+                    // TODO add checks or dataVersion
+                    if (path.clusterId === BasicInformationCluster.id && path.id !== BasicInformationCluster.attributes.nodeLabel.id) {
```

Comment:
Can you add more details why this specific attribute should be ignored?
Maybe better to split into 2 MRs: one that adds write support, the other that adds the hack for SmartThings

Reply:
Will split it

Reply:
But for context: smartthings sends a write request on basic info Cluster with * as attribute and a text value. According to specs we need to walk through all matching attributes and check the "constrains" and if "writable". Then in fact only the nodeLabel attribute is left to set that value on.

Because of the fact that I could not determine if an attribute is writable and it's length constrains I added that hack to only consider nodeLabel attribute when such a request comes in.

Reply:
Ok, let's implement checking if attributes are writable then.
I suggest above to add writable flag in the AttributeServer so we can create a getWritableAttributes method here that will returns only writable attributes.

Reply:
Good ... and how we handle the "constraints"? Reason: Fpr the Writecall that Smartthings is doing also "location" (2 char string) would be writable ... but should be ignored because contrains do not mauth to a "longer then 2 chars node label string" :-)

Reply:
Ok, I found the issue  I think .... will provide a PR to bring in validator logic

Reply:
https://github.com/mfucci/node-matter/pull/197

Reply:
and #198